### PR TITLE
feat(query): add TsCardinalities to SPP

### PR DESCRIPTION
**Pull request by alextheimer**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

Adds `TsCardinalities` `LogicalPlan` to `SinglePartitionPlanner`.

Additionally changes `groupDepth` to 1-indexed.

### CODE REVIEW

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ object LogicalPlanUtils extends StrictLogging {
       case lp: LabelValues                 => TimeRange(lp.startMs, lp.endMs)
       case lp: LabelCardinality            => TimeRange(lp.startMs, lp.endMs)
       case lp: LabelNames                  => TimeRange(lp.startMs, lp.endMs)
-      case lp: TsCardinalities             => throw new IllegalArgumentException("no time params for TsCardinalitites")
+      case lp: TsCardinalities             => TimeRange(0, Long.MaxValue)  // TODO(a_theimer)
```
Comment (alextheimer) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala:69]: HAP `materialize` always uses this function, so I need TsCardinalities to map to some `TimeRange` here. Is this sufficient?

Comment (alextheimer) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala]: no longer applicable

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala [hunk 2]
```diff
@@ -115,5 +117,11 @@ class SinglePartitionPlanner(planners: Map[String, QueryPlanner],
     if (execPlans.size == 1) execPlans.head
     else PartKeysDistConcatExec(qContext, inProcessPlanDispatcher, execPlans)
   }
+
+  private def materializeTsCardinalities(logicalPlan: TsCardinalities, qContext: QueryContext): ExecPlan = {
+    val execPlans = planners.values.toList.distinct.map(_.materialize(logicalPlan, qContext))
+    if (execPlans.size == 1) execPlans.head
+    else TsCardReduceExec(qContext, inProcessPlanDispatcher, execPlans)
```
Comment (vishramachandran) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala:124]: Can you change the logic to not materialize with every planner and pick first ? We can just materialize with first planner. I realize you picked this pattern from existing code. You can change the others.

**Review by vishramachandran — APPROVED** [review]

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 3]
```diff
@@ -110,7 +110,7 @@ object CliMain extends FilodbClusterNode {
     println("  --host <hostname/IP> [--port ...] --command status --dataset <dataset>")
     println("  --host <hostname/IP> [--port ...] --command labelvalues --labelnames <lable-names> --labelfilter <label-filter> --dataset <dataset>")
     println("  --host <hostname/IP> [--port ...] --command labels --labelfilter <label-filter> -dataset <dataset>")
-    println("  --host <hostname/IP> [--port ...] --command tscard --dataset <dataset> --shardkeyprefix <shard-key-prefix> --groupdepth {0,1,2}")
+    println("  --host <hostname/IP> [--port ...] --command tscard --dataset <dataset> --shardkeyprefix <shard-key-prefix> --groupdepth {1,2,3}")
```
[the remark this answers is not in the document]
Comment (alextheimer) [reply cli/src/main/scala/filodb.cli/CliMain.scala]: Done!

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala (hunk 2 above)
Comment (alextheimer) [reply coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala]: Updated; I assert the size is 1 and just return the single exec plan from the planner that matched the "raw" filter. (See the thread between Amol and I on Slack for additional discussion).

**Pull request merged**